test/src.test.ts

```diff
@@ -0,0 +1,244 @@
+import * as test from 'tape'
+import { applySelection, normalizeUrl, calculateTargetScore, getHTML, IPFS_GATEWAY } from '../src/main'
+import { BN } from 'bn.js'
+
+const whitelistedToken = '0x89d24A6b4CcB1B6fAA2625fE562bDD9a23260359'
+
+// Status is expired
+const ineligibleCampaign1 = {
+	depositAsset: whitelistedToken,
+	status: {
+		name: 'Expired'
+	},
+	spec: {
+		activeFrom: 0,
+		adUnits: [],
+		minPerImpression: new BN(2)
+	}
+}
+
+// Deposit asset not whitelisted
+const ineligibleCampaign2 = {
+	depositAsset: '0x0',
+	status: {
+		name: 'Active'
+	},
+	spec: {
+		activeFrom: 0,
+		adUnits: [],
+		minPerImpression: new BN(2)
+	}
+}
+
+// ActiveFrom has not commenced
+const ineligibleCampaign3 = {
+	depositAsset: whitelistedToken,
+	status: {
+		name: 'Active'
+	},
+	spec: {
+		activeFrom: Date.now() + 100000,
+		adUnits: [],
+		minPerImpression: new BN(2)
+	}
+}
+
+// No adUnits
+const ineligibleCampaign4 = {
+	depositAsset: whitelistedToken,
+	status: {
+		name: 'Active'
+	},
+	spec: {
+		activeFrom: 0,
+		minPerImpression: new BN(2)
+	}
+}
+
+// minPerImpression is too low
+const ineligibleCampaign5 = {
+	depositAsset: whitelistedToken,
+	status: {
+		name: 'Active'
+	},
+	spec: {
+		activeFrom: 0,
+		adUnits: [],
+		minPerImpression: new BN(0)
+	}
+}
+
+const options = {
+	publisherAddr: '0x13e72959d8055dafa6525050a8cc7c479d4c09a3',
+	width: 300,
+	height: 100,
+	marketURL: 'https://market.adex.network',
+	acceptedStates: ['Active', 'Ready'],
+	minPerImpression: new BN(1),
+	minTargetingScore: 0,
+	randomize: true,
+	whitelistedToken
+}
+
+const optionsWithTopByPrice = {
+	...options,
+	topByPrice: 1
+}
+
+const optionsWithTypeLimit = {
+	...options,
+	whitelistedType: 'legacy_300x100'
+}
+
+const optionsWithTargeting = {
+	...options,
+	targeting: [{tag: 'movies', score: 50}, {tag: 'music', score: 100}]
+}
+
+const optionsTargetingAndMinScore = {
+	...optionsWithTargeting
+}
+optionsTargetingAndMinScore.minTargetingScore = 1000
+
+const unit1 = {
```

Comment:
Why not do it like the options above.  It's the same units with just different targeting.